src/database/utility.cpp

```diff
@@ -1,7 +1,132 @@
+#include <stdio.h>
+#include <iostream>
+#include <vector>
+#include <fstream>
 #include "utility.h"
+#include "field.h"
+#include "heap_page.h"
 
 namespace buzzdb {
-/**
- * To be ported over
- */
+Utility::Utility() {}
+
+std::vector<Field::Type> Utility::GetTypes(int len) {
+  std::vector<Field::Type> types;
+  int idx = 0;
+  while (idx < len) {
+    types.push_back(Field::Type::INTEGER);
+    idx++;
+  }
+  return types;
+}
+
+std::vector<std::string> Utility::GetStrings(int len, std::string val) {
+  std::vector<std::string> strings;
+  int idx = 0;
+  while (idx < len) {
+    strings.push_back(val + std::to_string(idx++));
+  }
+  return strings;
+}
+
+TupleDesc Utility::GetTupleDesc(int n, std::string name) { 
+  return TupleDesc(GetTypes(n), GetStrings(n, name));
+}
+
+
+TupleDesc Utility::GetTupleDesc(int n) {
+  return TupleDesc(GetTypes(n));
+}
+
+Tuple Utility::GetHeapTuple(int n) {
+  Tuple tup = Tuple(GetTupleDesc(1));
+  HeapPageId heap_page_id(1, 2);
+  RecordId* rid = new RecordId(heap_page_id, 3);
+  tup.set_record_id(rid);
+
+  Field* field = new IntegerField(n);
+  tup.set_field(0, field);
+
+  delete rid;
```

Comment:
Hmmm, what are these deletes for?

If you delete the objects after you assigned them to the tuple, you will have undone what you just did earlier.